circe/src/main/scala/com/softwaremill/sttp/circe.scala

```diff
@@ -0,0 +1,32 @@
+package com.softwaremill.sttp
+
+import com.softwaremill.sttp.model.{
+  RequestBody,
+  ResponseAs,
+  SerializableBody,
+  StringBody
+}
+import io.circe.parser._
+import io.circe.{Decoder, Encoder}
+
+import scala.language.higherKinds
+
+package object circe {
+  private[sttp] val ApplicationJsonContentType = "application/json"
+
+  private def serializeBody[B](body: B)(
+      implicit encoder: Encoder[B]): RequestBody[Nothing] =
+    SerializableBody((b: B) => StringBody(encoder(b).noSpaces, Utf8), body)
+
+  implicit final class CirceRequestTOps[U[_], T, +S](val req: RequestT[U, T, S])
+      extends AnyVal {
+    def jsonBody[B: Encoder](body: B): RequestT[U, T, S] =
```

Comment:
Here's another approach, maybe worth considering: right now all the bodies are set via the `.body` overloaded method, with the exception of stream bodies (`.streamBody`), which has to be separate as it changes an additional type parameter on `RequestT`. There's also a variant of the `.body` method which allows any body as long at it serializes to a `BasicBody`:

```
type BodySerializer[B] = B => BasicRequestBody
```

Maybe this could be made into a `BodySerializer` as well? Maybe this would keep help the API more consistent?

One downside is that the default content type wouldn't be set. But maybe it would be possible to extend the `BasicRequestBody` with a `def defaultContentType: Option[String]` method, which would specify the content type to set, if there's none and one is defined?

It would then be the responsibility of the `BodySerializer` to return a `BasicRequestBody` with the appropriate data set (and that would be the responsibility of the library).

What do you think?